### ISSUE TITLE
PyDeadObject error after switching modes?

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#697]
_[Trac time 20090612 174739Z]_ I'm getting a PyDeadObjectError:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14574, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/work/src/parthenon/flaash/flaash_mode.py", line 1783, in SetFocus
    page = self.GetCurrentPage()
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14524, in __getattr__
    raise PyDeadObjectError(self.attrStr % self._name)
PyDeadObjectError: The C++ part of the FlaashMode object has been deleted, attribute access no longer allowed.
```

which I've not yet been able to reliably duplicate, but occurred most recently when changing the major mode from something to FundamentalMode.

**Comment by robmcmullen** [comment]
_[Trac time 20090612 174839Z]_ It's coming from a CallAfter, that much is clear -- but because it is, it's not possible to see which CallAfter is the culprit.

**Comment by robmcmullen** [comment]
_[Trac time 20090613 144003Z]_ Remembered the CallAfter debugging code in main and found:

```
main.py:35 NewCallAfter: CallAfter: args=(<bound method BufferFrame.isModeChangedOnDisk of <peppy.frame.BufferFrame; proxy of <Swig Object of type 'wxFrame *' at 0x2ddaed0> >>,) kwargs={}
main.py:35 NewCallAfter: CallAfter: args=(<bound method FlaashMode.SetFocus of <flaash.flaash_mode.FlaashMode; proxy of <Swig Object of type 'wxNotebook *' at 0x2f3c1b0> >>,) kwargs={}
main.py:35 NewCallAfter: CallAfter: args=(<bound method MajorModeSelect.action of <peppy.mainmenu.MajorModeSelect object at 0x3062990>>, 9, 1) kwargs={}
main.py:35 NewCallAfter: CallAfter: args=(<bound method FlaashMode.SetFocus of <flaash.flaash_mode.FlaashMode; proxy of <Swig Object of type 'wxNotebook *' at 0x2f3c1b0> >>,) kwargs={}
major.py:500 getClassCache: Creating class cache for <class 'peppy.fundamental.FundamentalMode'>
multikey.py:598 UserAccelerators._addKeyBinding: C-l already bound to menu item RecenterScreen and takes precedence over attempted rebind to CutLine
Traceback (most recent call last):
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14574, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/work/src/parthenon/flaash/flaash_mode.py", line 1789, in SetFocus
    page = self.GetCurrentPage()
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14524, in __getattr__
    raise PyDeadObjectError(self.attrStr % self._name)
PyDeadObjectError: The C++ part of the FlaashMode object has been deleted, attribute access no longer allowed.
```

**Comment by robmcmullen** [comment]
_[Trac time 20090613 144642Z]_ Found the culprit CallAfter in flaash mode.

**Comment by robmcmullen** [comment]
_[Trac time 20090613 154510Z]_ c482a994d736ebf73be882fb544ac534bf1ecfac Fixed #697: added traceback to CallAfter debug printing
